cdcv2/metadata/metadata.go

```diff
@@ -42,30 +73,27 @@ type CaptureObservation interface {
 	) error
 
 	// Advance advances some changefeed progresses that are collected from processors.
-	Advance(cfs []ChangefeedIDWithEpoch, progresses []ChangefeedProgress) error
+	Advance(cp CaptureProgress) error
 
 	// Fetch owner modifications.
 	OwnerChanges() <-chan ScheduledChangefeed
 
 	// When an owner exits, inform the metadata storage.
-	PostOwnerRemoved(cf ChangefeedIDWithEpoch) error
-
-	// Fetch processor list modifications.
-	ProcessorChanges() <-chan ScheduledChangefeed
-
-	// When a processor exits, inform the metadata storage.
-	PostProcessorRemoved(cf ChangefeedIDWithEpoch) error
+	PostOwnerRemoved(cf ChangefeedUUID) error
 }
 
 // ControllerObservation is for observing and updating meta by Controller.
 //
 // All intrefaces are thread-safe and shares one same Context.
 type ControllerObservation interface {
-	// CreateChangefeed creates a changefeed, Epoch will be filled into the input ChangefeedInfo.
-	CreateChangefeed(cf *ChangefeedInfo, up *model.UpstreamInfo) (ChangefeedIDWithEpoch, error)
+	// CreateChangefeed creates a changefeed, UUID will be filled into the input ChangefeedInfo.
+	CreateChangefeed(cf *ChangefeedInfo, up *model.UpstreamInfo) (ChangefeedIdent, error)
```

Comment:
add a context parameter?

Reply:
It is important to consider whether ctx or timeout is needed. The difference is that using ctx implies semantically that this function may start goroutine, while passing timeout only means that we want a blocking operation to end within a specified time. From my personal point of view, timeout might be more appropriate, because all goroutine resources are bound to the observation.